corehq/pillows/groups_to_user.py

```diff
@@ -67,16 +68,29 @@ def update_es_user_with_groups(group_doc, es_client=None):
             doc = {"__group_ids": list(user_source.group_ids), "__group_names": list(user_source.group_names)}
             es_client.update(USER_INDEX, ES_META['users'].type, user_source.user_id, body={"doc": doc})
 
+    for user_source in stream_user_sources(group_doc.get("removed_users", [])):
+        if group_doc["name"] in user_source.group_names or group_doc["_id"] in user_source.group_ids:
+            user_source.group_ids.remove(group_doc["_id"])
+            user_source.group_names.remove(group_doc["name"])
+            doc = {"__group_ids": list(user_source.group_ids), "__group_names": list(user_source.group_names)}
+            es_client.update(USER_INDEX, ES_META['users'].type, user_source.user_id, body={"doc": doc})
+
 
 UserSource = namedtuple('UserSource', ['user_id', 'group_ids', 'group_names'])
 
 
 def stream_user_sources(user_ids):
-    q = {"filter": {"and": [{"terms": {"_id": user_ids}}]}}
-    for result in stream_es_query(es_index='users', q=q, fields=["__group_ids", "__group_names"]):
-        group_ids = result.get('fields', {}).get("__group_ids", [])
+    results = (
+        UserES()
+        .user_ids(user_ids)
+        .fields(['_id', '__group_ids', '__group_names'])
+        .scroll()
+    )
```

Comment:
note this query is _slightly_ different from the original. this one excludes deleted and inactive users, which i think is fine, but maybe for consistency sake we should keep them up to date?